src/greenlight/runner.clj

```diff
@@ -38,22 +44,41 @@
     (report/write-html-results html-path results nil)))
 
 
+(defn- filter-tests
+  [matcher test-vars]
+  (cond->> test-vars
+    (keyword? matcher)
+    (filter (comp matcher meta))
+
+    (instance? java.util.regex.Pattern matcher)
+    (filter #(re-matches matcher (name (:name (meta %)))))
+
+    true
+    (map (fn [v] (v)))
```

Comment:
This part seems a little non-obvious to me - is it calling the test step constructor so that you can match against the produced attributes in the next test?

Reply:
We were previously finding test vars and only filtering on metadata of those vars, then calling the test constructor after the filter. 

https://github.com/amperity/greenlight/blob/38898c149b3c723475d20a506546fd4984682a5f/src/greenlight/runner.clj#L53-L56

I wanted to extend that ability to also select on test properties (specifically the `::test/group`), but that requires calling the test constructor and then doing some additional filtering. 

I guess worst case we end up calling the test constructor (which could have side effects?) on tests that will not be selected.